### PR TITLE
[ENG-4034] adapt NotableEmailDomain model

### PR DESCRIPTION
**Pull request by adlius**
## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

File: osf/models/notable_domain.py [hunk 1]
```diff
@@ -6,10 +6,12 @@
 from osf.utils.fields import LowercaseCharField
 
 
-class NotableEmailDomain(BaseModel):
+class NotableDomain(BaseModel):
     class Note(IntEnum):
-        EXCLUDE_FROM_ACCOUNT_CREATION = 0
+        EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT = 0
         ASSUME_HAM_UNTIL_REPORTED = 1
+        UNKOWN = 2
```
Comment (Johnetordoff) [inline osf/models/notable_domain.py:13]: nit: typo here

**Comment by jwalz** [comment]
Please rename the auto migration with a descriptive name (this is a thing I'd like to start doing that we haven't always done in the past). Other than that, my *only* thoughts is merging this with the PR to add the DomainReference model so that we can limit our migration count.

File: osf/models/notable_domain.py [hunk 2]
```diff
@@ -6,10 +6,12 @@
...
 
 
-class NotableEmailDomain(BaseModel):
+class NotableDomain(BaseModel):
     class Note(IntEnum):
-        EXCLUDE_FROM_ACCOUNT_CREATION = 0
+        EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT = 0
         ASSUME_HAM_UNTIL_REPORTED = 1
+        UNKNOWN = 2
+        IGNORED = 3
```
Comment (aaxelb) [inline osf/models/notable_domain.py:14]: so is that a no on `Clam`?
```
SPAM = 0     # bad
HAM = 1      # good
CLAM = 2     # probably fine, but sniff it first
UNKNOWN = 3  # hasn't yet been categorized
```

**Review by aaxelb — APPROVED** [review]
🎶

**Comment by adlius** [comment]
Close in favor of #10063

**Pull request closed without merging**